source/slang/slang-ir.h

```diff
@@ -733,6 +733,11 @@ struct IRInst
         getOperands()[index].init(this, value);
     }
 
+    // Reserved memory space for use by individual IR passes.
```

Comment:
64 bits is quite a lot of bits.
If you wanted a boolean test, you wouldn't need a reset if a counter was used. Each new used an increment of the counter, and a inst is reset if the scratch isn't equal to the counter. 
If you want multiple bits that doesn't work.

Reply:
This is to allow potential case where a pass may want to stick a pointer here for general data.

Reply:
The current design is that an IR pass should not make  any assumptions on the intiial value of scratchData, and no conventions should be created to coordinate the access of the field across passes to prevent hard to debug issues. It is therefore safer to not use the counter strategy for now.